modules/videoio/test/test_gstreamer.cpp

```diff
@@ -0,0 +1,71 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+#include "opencv2/videoio/videoio_c.h"
+#ifdef HAVE_GSTREAMER
+
+namespace opencv_test
+{
+typedef tuple< string, Size, Size, int > Param;
+typedef testing::TestWithParam< Param > Videoio_Gstreamer_Test;
+
+TEST_P(Videoio_Gstreamer_Test, test_object_structure)
+{
+    string format    = get<0>(GetParam());
+    Size frame_size  = get<1>(GetParam());
+    Size mat_size    = get<2>(GetParam());
+    int convertToRGB = get<3>(GetParam());
+    int count_frames = 10;
+    std::ostringstream pipeline;
+    pipeline << "videotestsrc pattern=ball num-buffers=" << count_frames << " ! " << format;
+    pipeline << ", framerate=1/1, width=" << frame_size.width << ", height=" << frame_size.height << " ! appsink";
+    VideoCapture cap(pipeline.str(), CAP_GSTREAMER);
+
+    ASSERT_TRUE(cap.isOpened());
+    for (int i = 0; i < count_frames; ++i)
+    {
+        Mat buffer, frame, gray_frame, rgb_frame, binary_frame;
```

Comment:
All these variables can be moved out of the loop body, unused variables should be removed.